_aicspylibczi/SourceRange.h

```diff
@@ -11,88 +11,85 @@
-              return *this;
-          }
-
-          SourceChannelIterator operator++(int)
-          {
-              SourceChannelIterator preIncrementIterator = *this;
-              ++(*this);
-              return preIncrementIterator;
-          }
-
-          bool operator==(const SourceChannelIterator& other_) const
-          {
-              return *(m_channelIterators.begin())==*(other_.m_channelIterators.begin());
-          }
-
-          bool operator!=(const SourceChannelIterator& other_) const
-          {
-              return !(*this==other_);
-          }
-
-          std::vector<T*> &operator*()
-          {
-              return m_channelIterators;
-          }
-          // iterator traits
-          using difference_type = size_t;
-          using value_type = T;
-          using pointer = T*;
-          using reference = T&;
-          using iterator_category = std::forward_iterator_tag;
-      };
-
-      SourceChannelIterator begin()
-      {
-          return SourceChannelIterator(m_channels, m_begin);
-      }
-
-      SourceChannelIterator strideBegin(size_t y_index_)
-      {
-          return SourceChannelIterator(m_channels, (T*) (((uint8_t*) m_begin)+y_index_*m_stride));
-      }
-
-      SourceChannelIterator strideEnd(size_t y_index_)
-      {
-          auto tmp = (uint8_t*) m_begin;
-          tmp += y_index_*m_stride+m_pixelsPerStride*m_channels*sizeof(T);
-          T* sEnd = (T*) tmp;
-          if (sEnd>m_end)
-              throw ImageIteratorException(
-                  "stride advanced pointer beyond end of array.");
-          return SourceChannelIterator(m_channels, sEnd);
-      }
-
-      SourceChannelIterator end() { return SourceChannelIterator(m_channels, m_end); }
+    SourceChannelIterator(size_t number_of_channels_, T* ptr_)
+      : m_channelIterators(number_of_channels_)
+    {
+      std::generate(m_channelIterators.begin(), m_channelIterators.end(), [ptr_]() mutable { return ptr_++; });
+    }
+
+    SourceChannelIterator& operator++()
+    {
+      size_t numberOfChannels = m_channelIterators.size();
+      std::for_each(m_channelIterators.begin(), m_channelIterators.end(), [numberOfChannels](T*& p_) {
+        p_ = p_ + numberOfChannels;
+      });
+      return *this;
+    }
+
+    SourceChannelIterator operator++(int)
+    {
+      SourceChannelIterator preIncrementIterator = *this;
+      ++(*this);
+      return preIncrementIterator;
+    }
+
+    bool operator==(const SourceChannelIterator& other_) const
+    {
+      return *(m_channelIterators.begin()) == *(other_.m_channelIterators.begin());
+    }
+
+    bool operator!=(const SourceChannelIterator& other_) const { return !(*this == other_); }
+
+    std::vector<T*>& operator*() { return m_channelIterators; }
+    // iterator traits
+    using difference_type = size_t;
+    using value_type = T;
+    using pointer = T*;
+    using reference = T&;
+    using iterator_category = std::forward_iterator_tag;
```

Comment:
is this all just reformatting or did something change?  is it in scope or out of scope of this pr?

Reply:
It's just reformatting. I thought I would have to change class at one point but this code is not actually used anymore due to it now just using memcpy

Reply:
Then it can be removed?